### ISSUE TITLE
"call: command not found" in .sh build script

### ISSUE THREAD

**Issue by asaf92** [dotnet/BenchmarkDotNet#1753]
I'm trying to run BenchmarkDotNet in Rider IDE on Ubuntu with .NET5. 

I always get this error:
```
// Found 1 benchmarks:
//   MyBenchmarkClass.Foo: DefaultJob

// Build Error: Standard output: 
 Caused by: java.lang.ClassNotFoundException: restore
Error: Could not find or load main class restore 
 Standard error: 
 

// BenchmarkDotNet has failed to build the auto-generated boilerplate code.
// It can be found in /home/.../BenchmarksExample/bin/Release/net5.0/d3a63bcd-c17f-47fd-b4c3-9078cfdfe110
// Please follow the troubleshooting guide: https://benchmarkdotnet.org/articles/guides/troubleshooting.html
// ***** BenchmarkRunner: Finish  *****

```

I went through the troubleshooting page and it said to go to the output folder and:

> Run the (.sh) script, read the error message. From here you continue with the troubleshooting like it was a project in your solution.

This is the script:
```
call dotnet restore  /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true
call dotnet build -c Release  /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true
```

When I change the permissions and run the script, I get this error:
```
./d3a63bcd-c17f-47fd-b4c3-9078cfdfe110.sh: line 1: call: command not found
./d3a63bcd-c17f-47fd-b4c3-9078cfdfe110.sh: line 2: call: command not found

```
If I remove the `call` parts, it builds successfully.

How do I get BenchmarkDotNet to work?

**Comment by adamsitnik** [comment]
@asaf92 Hi! What command do you use to run the benchmarks from Rider IDE? I just tried running `dotnet run -c Release` from `Terminal` window and it works fine for me.

**Comment by asaf92** [comment]
I just run the console application with Ctrl+F5.
Turns out `dotnet` commands actually work but running the console app doesn't. 

Is that expected?

**Comment by adamsitnik** [comment]
I always run the benchmarks from terminal to make sure benchmarks and IDE are not competing for resources

> Is that expected?

@AndreyAkinshin who work on Rider IDE is the best person to answer this question.

**Comment by asaf92** [comment]
There's a workaround:
Add "--cli /usr/bin/dotnet" (or whatever the path to the dotnet executable is) to "Program arguments" in the "Run/Debug Configurations" window.

**Comment by AndreyAkinshin** [comment]
OK, I have found the bug. Here is our current code (BenchmarkDotNet v0.13.1) that detects the proper `dotnet` location:

```cs
private static string GetDefaultDotNetCliPath()
{
    if (!Portability.RuntimeInformation.IsLinux())
        return "dotnet";

    using (var parentProcess = Process.GetProcessById(getppid()))
    {
        string parentPath = parentProcess.MainModule?.FileName ?? string.Empty;
        // sth like /snap/dotnet-sdk/112/dotnet and we should use the exact path instead of just "dotnet"
        if (parentPath.StartsWith("/snap/", StringComparison.Ordinal))
        {
            return parentPath;
        }

        return "dotnet";
    }
}

[DllImport("libc")]
private static extern int getppid();
```

The "snap" part of `GetDefaultDotNetCliPath` was implemented in the scope of the basic snap support (see #1570, #1652). The above approach works great when you run benchmarks via `dotnet run`. However, Rider builds an application in advance and just calling the executable file. The parent process of the running process is the main Rider process which is `java`. If you install Rider via snap (@asaf92 am I right that you installed Rider via snap?), the java executable will be detected as the default dotnet runtime.

I have fixed this bug in 1a8296f86ead19359c842a36ebd22b223695afa2, the fix will be available in the next BenchmarkDotNet release (presumably v0.13.2).
@asaf92 your workaround with `"--cli /usr/bin/dotnet"` is absolutely correct since it overrides the default dotnet location with the given one.

**Comment by asaf92** [comment]
Thanks @AndreyAkinshin ! 
You are correct, I did install Rider via snap. 

In regards to the workaround - notice that I only found this workaround by running the benchmarks in debug mode (or with the debugger attached maybe? can't remember) and saw an error message hinting about the `--cli` usage. It did not show when I ran the benchmarks normally in Release mode, so perhaps the error message not showing is also an issue?
